torch/testing/_internal/distributed/multi_threaded_pg.py

```diff
@@ -369,6 +370,8 @@ def __repr__(self):
 
 
 def _create_threaded_pg(prefix_store, rank, world_size, timeout):
+    # TODO: we need to find a way to work around this properly for mtpg.
+    os.environ["TORCH_DIST_INIT_BARRIER"] = "1"
```

Comment:
This env var isn't unset so I think as a side effect it would also turn on the barrier for any subsequent tests (even non-mtpg ones). We can either

1. set and unset this in MultiThreadedTestcase cleanup / teardown
2. manually add a barrier after init_process_group("threaded", ...) in our mtpg tests